src/components/dialogs/network-modifications/tabular-modification/tabular-modification-form.tsx

```diff
@@ -0,0 +1,199 @@
+ */
+
+import Grid from '@mui/material/Grid';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { useFormContext, useWatch } from 'react-hook-form';
+import { AutocompleteInput } from '@gridsuite/commons-ui';
+import { gridItem } from 'components/dialogs/dialogUtils';
+import { MODIFICATIONS_TABLE, TYPE } from 'components/utils/field-constants';
+import { EQUIPMENT_TYPES } from 'components/utils/equipment-types';
+import { useCSVPicker } from 'components/utils/inputs/input-hooks';
+import CsvDownloader from 'react-csv-downloader';
+import { Alert, Button } from '@mui/material';
+import {
+    Modification,
+    TABULAR_MODIFICATION_FIELDS,
+    styles,
+} from './tabular-modification-utils';
+import { CustomAGGrid } from 'components/custom-aggrid/custom-aggrid';
+import { DefaultCellRenderer } from 'components/spreadsheet/utils/cell-renderers';
+import Papa from 'papaparse';
+
+const richTypeEquals = (a: any, b: any) => a === b;
+
+const TabularModificationForm = () => {
+    const intl = useIntl();
+
+    const { setValue } = useFormContext();
+
+    const richTypeLabel = (rt: any) => {
+        return intl.formatMessage({ id: rt });
+    };
+
+    const watchType = useWatch({
+        name: TYPE,
+    });
+
+    const csvColumns = useMemo(() => {
+        return TABULAR_MODIFICATION_FIELDS[watchType]?.map((field) => {
+            return intl.formatMessage({ id: field });
+        });
+    }, [intl, watchType]);
+
+    const [typeChanged, setTypeChanged] = useState(false);
+    const [selectedFile, FileField, selectedFileError] = useCSVPicker({
+        label: 'ImportModifications',
+        header: csvColumns,
+        maxTapNumber: undefined,
+        disabled: !csvColumns,
+        resetTrigger: typeChanged,
+    });
+
+    const postProcessFile = useCallback(
+        (fileData: any) => {
+            fileData = fileData.map((row: any) => {
+                let modification: Modification = {};
+                Object.getOwnPropertyNames(row).forEach((key) => {
+                    modification[key] = row[key];
+                });
+                return modification;
+            });
+            setValue(MODIFICATIONS_TABLE, fileData, { shouldDirty: true });
+        },
+        [setValue]
+    );
+
+    const watchTable = useWatch({
+        name: MODIFICATIONS_TABLE,
+    });
+
+    useEffect(() => {
+        if (selectedFileError) {
+            setValue(MODIFICATIONS_TABLE, []);
+        } else if (selectedFile) {
+            Papa.parse(selectedFile as any, {
+                header: true,
+                skipEmptyLines: true,
+                complete: function (results) {
+                    postProcessFile(results.data);
+                },
+            });
+        }
+    }, [postProcessFile, selectedFile, selectedFileError, setValue]);
+
+    const typesOptions = useMemo(() => {
+        const equipmentTypesToExclude = new Set([
```

Comment:
do you thing that is necessary to use a Set? 
then you can use includes in place of has